src/java.desktop/windows/native/libawt/windows/awt_TrayIcon.cpp

```diff
@@ -221,6 +221,26 @@ void AwtTrayIcon::InitNID(UINT uID)
     m_nid.uVersion = NOTIFYICON_VERSION;
 }
 
+// trayicon update handler called when Screen scale changes
+void AwtTrayIcon::UpdateTrayIconHandler()
+{
+    jmethodID updateTrayFn;
+    jclass systemTrayClass;
+
+    JNIEnv *env =(JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
+    systemTrayClass = env->FindClass("java/awt/SystemTray");
+
+    if (systemTrayClass != NULL) {
+         updateTrayFn = env->GetStaticMethodID(
+                     systemTrayClass, "updateTrayIcons", "()V");
+         if (updateTrayFn != NULL) {
+             env->CallStaticVoidMethod(systemTrayClass,
+                      updateTrayFn);
```

Comment:
Most of these JNI methods can leave an exception pending
https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#findclass
https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#getstaticmethodid
https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#callstatictypemethod-routines-callstatictypemethoda-routines-callstatictypemethodv-routines

I think you should be calling ExceptionClear() because subsequent methods should not be called with an exception pending. And this is a windows thread attached to the VM and it never "returns to Java" but will be called back to handle later events.